hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -205,28 +206,17 @@ checkCollectCom ctx@ScriptContext{scriptContextTxInfo = txInfo} (contestationPer
   -- Collect fuel and commits from resolved inputs. Any output containing a PT
   -- is treated as a commit, "our" output is the head output and all remaining
   -- will be accumulated as 'fuel'.
-  traverseInputs (fuel, commits, nCommits) = \case
-    [] ->
+  traverseInputs TxInInfo{txInInfoResolved} (fuel, commits, nCommits)
```

Comment:
This function does not traverse all the inputs anymore, I would, either, rename the function to make that clear but I'm not sure:

```suggestion
  traverseInput TxInInfo{txInInfoResolved} (fuel, commits, nCommits)
```

Or I would keep the function's name and keep the call site as it was by introducing some auxiliary function to traverseInputs:
```suggestion
  traverseInputs someList  =
    foldr auxiliary someList
  where
   auxiliary TxInInfo{txInInfoResolved} (fuel, commits, nCommits) =
```

Reply:
`extractAndCountCommits` it is! :)